litellm/router.py

```diff
@@ -2089,6 +2090,10 @@ def set_client(self, model: dict):
                     raise ValueError(
                         f"api_base is required for Azure OpenAI. Set it on your config. Model - {model}"
                     )
+                azure_ad_token = litellm_params.get("azure_ad_token")
+                if azure_ad_token is not None:
+                    if azure_ad_token.startswith("oidc/"):
```

Comment:
@Manouchehri can we add a simple unit test in `test_router.py` to make sure this is always respected? 

We've had previous regressions due to untested flows (thinking of cloudflare). Would help to have something for this